### ISSUE TITLE
Elevation profile for routes: Make "Show Elevation profile" more prominent

### ISSUE THREAD

**Issue by MagpieFourtyTwo** [cgeo/cgeo#15707]
### Tell us your idea!

I *knew* it exists, but was not able to find it for about half a year now - finally even thought I just fantasized it. until I accidentally found it again this weekend: The elevation profile for individual routes.  ;)

Just in case there are some more who do not know how to get it:
1. you need an individual route which contains a cache as routing point
2. long press click this cache routing point in the exact correct way (which is already hard enough, believe me ;) ) 
3. if you did it right, the ``Select item`` popup will contain the cache plus an entry called ``Individual route``
4. click this ``Individual route`` entry and *yeah* there you are: The elevation profile will finally pop up.  ;)

Most astonishingly this select menu does not always appear.  Most of the times the route edit popup appears - did not yet manage to understand the rules ... and even if you get the ``Individual route`` entry, this does not *always* mean you will get the elevation profile. Like now ... tried to get it just for testing, but could manage to get it ... so even now that I thought I am pretty sure to know how to get it, I can't. Frustrating.

### What solution would you suggest?

- Add ``Show elevation profile`` (resp. an appropriate icon) to the route's context/edit menu
- Add ``Show elevation profile`` to the route's bottom sheet
- Moreover I would highly appreciate an elevation profile for "normal" routes, too  :)
- Perhaps one has a deterministic way to get the profile even now ...?  ;)

**Comment by ztNFny** [comment]
Just a short tap on the route opens the elevation view as well.

**Comment by MagpieFourtyTwo** [comment]
That's good to know - thanks.

However, I wanted to show this really useful feature (especially for bike cachers) to some people at the event tonight, but failed completely. So, are there any additional requirements that need to be met for this function to be displayed?  And which ones?

**Comment by moving-bits** [comment]
Maybe this?
It's available on UnifiedMap only

**Comment by MagpieFourtyTwo** [comment]
Good idea - did not check ... and at least one device that was the reason. :) THX!

**Comment by moving-bits** [comment]
NB: With #16105 opening elevation chart is moved to the long-tap context menu for individual route / route / track.
IMHO we can close this issue here then.

**Comment by MagpieFourtyTwo** [comment]
Till now I did not even know (or had simply forgotten) that there is a long click context menu for individual routes at all. :p

I (and most of the guys I know) usually use the bottom sheet menu (which is pretty obvious thanks to the additional symbol).

Any chance to add a symbol for the elevation chart over there, too?

**Comment by moving-bits** [comment]
The info sheet already has quite a lot of buttons per route/track:

Individual route has five buttons (but a relatively short name), whereas routes/tracks from GPX have four buttons, but potentially longer names). IMHO we should not add further buttons here.

![image](https://github.com/user-attachments/assets/1af28700-3944-4a6b-9add-0f0ad4ddb01b)

**Comment by moving-bits** [comment]
To implement further actions here, we could use an "overflow menu like approach", meaning: replace one of the existing buttons with an overflow symbol ("hamburger" icon), which then opens a popup with the additional actions.

I propose to create an overflow menu with the following items:
- delete track/route
- show elevation chart
- (as well as additional items for individual routes)

So basically use the items the long tap menu for track/route currently has, but show four buttons at most, and act like having the overflow menu mechanism an action bar has.

**Comment by ztNFny** [comment]
I propose to keep edit, delete, show/hide (in order of priority) as default options (outside overflow menu) as I think those are the most common ones.

Would also suggest to change the tooltip of "refresh" to "refresh caches" - until I just tried it I had no idea what that button even does - having it in overflow menu with text would make it even more discoverable 

And could add "optimize" to the overflow menu instead of just in edit dialog.

**Comment by moving-bits** [comment]
> I propose to keep edit, delete, show/hide (in order of priority) as default options (outside overflow menu) as I think those are the most common ones.

"Color" should stay there as well (for tracks) to make it easier to distinguish tracks if more than one is loaded.

I proposed "delete" to go to the overflow menu as I need one button to go, and as I regard this one as less frequently used (yet to be proven, though), and moving it to the overflow would be sort of an additional safety barrier. But TBD.

Thus my proposal would be
|first colum|second|third|fourth|fifth|
|---|---|---|---|---|
|Name|edit / color|hide/unhide|focus|overflow|


> Would also suggest to change the tooltip of "refresh" to "refresh caches" - until I just tried it I had no idea what that button even does - having it in overflow menu with text would make it even more discoverable

agreed

> And could add "optimize" to the overflow menu instead of just in edit dialog.

this would be part of "as well as additional items for individual routes" :-)

I would also remove it from edit dialog then - no longer needed over there, IMHO.

**Comment by ztNFny** [comment]
> "Color" should stay there as well (for tracks) to make it easier to distinguish tracks if more than one is loaded.

Color of course, I was only mentioning the buttons.
 
> I proposed "delete" to go to the overflow menu as I need one button to go, and as I regard this one as less frequently used (yet to be proven, though

I've never used focus and very very rarely hide/show, but I use delete all the time ;-) Actually I also never use color ... delete is 90% and edit 10% of my use of those.

> I would also remove it from edit dialog then - no longer needed over there, IMHO.

Wouldn't do that, it's quite related to editing.

**Comment by moving-bits** [comment]
Some progress over here:

I've exchanged the manually defined items in routes/tracks quick settings by toolbars, treating the items as a regular menu. This allows me to use identical menus for quick settings, long tap context menu and elevation chart popup menu, which leads to some functional extensions.

Some open todo are:
- get rid of the extra indentation in quick settings toolbars (all values are already set to 0, don't know where the remaining margin/paadding comes from)
- replace generic color palette icon by the specific color set for the track for easier recognition
- get rid of duplicated code (this is WIP currently)

|quick settings|route overflow menu|track overflow menu|
|---|---|---|
|![image](https://github.com/user-attachments/assets/c65b9b76-9142-4db5-b6f8-7b7e405312bb)|![image](https://github.com/user-attachments/assets/6b128ab7-8091-4682-a953-3c3972eca57e)|![image](https://github.com/user-attachments/assets/004225ad-3f04-4e36-8092-4c02c7b31d9c)|

|long tap on route in map|elevation chart menu for route|
|---|---|
|![image](https://github.com/user-attachments/assets/7d82cd44-0cd1-47fa-ab21-617a311f6e39)|![image](https://github.com/user-attachments/assets/da763de8-69ac-4667-a664-246e01fd7d42)|

**Comment by ztNFny** [comment]
I'm a bit confused by the colour palette. In your screenshot i see a generic palette button while before it was a square of the selected colour?

**Comment by ztNFny** [comment]
> I'm a bit confused by the colour palette. In your screenshot i see a generic palette button while before it was a square of the selected colour?

Never mind, i missed that it's in todo section. Too late ...

Could you enable icons for those overflow menus?

**Comment by moving-bits** [comment]
> Could you enable icons for those overflow menus?

I'll add this to the todo list. It's a bit hacky, IIRC.

**Comment by ztNFny** [comment]
> > Could you enable icons for those overflow menus?
> 
> I'll add this to the todo list. It's a bit hacky, IIRC.

It's a setting on popupmenu, i did it for the map selection menu a while ago.

**Comment by ztNFny** [comment]
![Screenshot_2025-01-06-21-23-19-10_ffb2f5e1b976ff98cfc94f359fbce8de.jpg](https://github.com/user-attachments/assets/18ec1324-7fe5-45f7-b0b2-ac64e569e0c0)

**Comment by moving-bits** [comment]
> > > Could you enable icons for those overflow menus?
> > 
> > 
> > I'll add this to the todo list. It's a bit hacky, IIRC.
> 
> It's a setting on popupmenu, i did it for the map selection menu a while ago.

thanks, but toolbar's overflow menu does not inherit from `PopupMenu`. But I applied the same hack I'm already using for the ElevationChart's overflow menu:

```
@SuppressLint("RestrictedApi") // required for workaround to make icons visible in overflow menu
[...]
if (menu instanceof MenuBuilder) {
    ((MenuBuilder) menu).setOptionalIconsVisible(true);
}
```

**Comment by ztNFny** [comment]
In that toolbar menu it's also not inheriting from PopupMenu but replacing the default one with a completly custom menu there using PopupMenu - iirc it was even you who introduced that?

**Comment by moving-bits** [comment]
See PR #16531

**Comment by moving-bits** [comment]
Implemented with #16531, so let's close here
